src/dragonball/src/dbs_virtio_devices/src/vhost/vhost_user/block.rs

```diff
@@ -0,0 +1,776 @@
+        trace!(target: "vhost-blk", "{}: VirtioDevice::write_config(0x{:x}, {:?})",
+               self.id, offset, data);
+
+        self.device().device_info.write_config(offset, data)
+    }
+
+    fn activate(&mut self, config: device::VirtioDeviceConfig<AS, Q, R>) -> ActivateResult {
+        trace!(target: "vhost-blk", "{}: VirtioDevice::activate()", self.id);
+
+        let mut device = self.device();
+        if config.queues.len() != device.queue_sizes.len() {
+            error!(
+                "{}: cannot perform activate, expected {} queue(s), got {}",
+                self.id,
+                device.queue_sizes.len(),
+                config.queues.len()
+            );
+            return Err(ActivateError::InvalidParam);
+        }
+
+        let timer_fd = device.timer_fd.take().unwrap();
+        let handler = VhostUserBlockHandler {
+            device: self.device.clone(),
+            queue_sizes: self.queue_sizes.clone(),
+            intr_evts: device.intr_evts.clone(),
+            timer_fd,
+            config,
+            id: self.id.clone(),
+        };
+
+        device.setup_slave(&handler)?;
+        let epoll_mgr = device.device_info.epoll_manager.clone();
+        drop(device);
+        self.subscriber_id = Some(epoll_mgr.add_subscriber(Box::new(handler)));
+
+        Ok(())
+    }
+
+    fn get_resource_requirements(
+        &self,
+        requests: &mut Vec<ResourceConstraint>,
+        use_generic_irq: bool,
+    ) {
+        trace!(target: "vhost-blk", "{}: VirtioDevice::get_resource_requirements()", self.id);
+
+        requests.push(ResourceConstraint::LegacyIrq { irq: None });
+        if use_generic_irq {
+            /* Allocate one irq for device configuration change events, and one irq for each queue. */
+            requests.push(ResourceConstraint::GenericIrq {
+                size: (self.queue_sizes.len() + 1) as u32,
+            });
+        }
+    }
+
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn as_any_mut(&mut self) -> &mut dyn Any {
+        self
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::{mem, sync::Arc, thread, time::Duration};
+
+    use dbs_device::resources::DeviceResources;
+    use dbs_interrupt::{InterruptManager, InterruptSourceType, MsiNotifier, NoopNotifier};
+    use dbs_utils::epoll_manager::EpollManager;
+
+    use kvm_ioctls::Kvm;
+
+    use vhost_rs::vhost_user::message::{
+        VhostUserConfig, VhostUserProtocolFeatures, VhostUserU64, VhostUserVirtioFeatures,
+    };
+    use vhost_rs::vhost_user::Listener;
+    use virtio_queue::QueueSync;
+    use vm_memory::{FileOffset, GuestMemoryMmap, GuestRegionMmap};
+    use vmm_sys_util::tempfile::TempFile;
+
+    use crate::vhost::vhost_user::block::{
+        VhostUserBlock, VhostUserConfigFlags, VirtioBlockConfig, VHOST_USER_CONFIG_OFFSET,
+    };
+    use crate::vhost::vhost_user::test_utils::*;
+    use crate::{GuestAddress, VirtioDevice, VirtioDeviceConfig, VirtioQueueConfig, TYPE_BLOCK};
```

Comment:
1. No extra blanks in the part of other crates.
2. Split use blocks that have different upper packages.

```suggestion
    use std::{mem, thread};
    use std::sync::Arc;
    use std::time::Duration;

    use dbs_device::resources::DeviceResources;
    use dbs_interrupt::{InterruptManager, InterruptSourceType, MsiNotifier, NoopNotifier};
    use dbs_utils::epoll_manager::EpollManager;
    use kvm_ioctls::Kvm;
    use vhost_rs::vhost_user::message::{
        VhostUserConfig, VhostUserProtocolFeatures, VhostUserU64, VhostUserVirtioFeatures,
    };
    use vhost_rs::vhost_user::Listener;
    use virtio_queue::QueueSync;
    use vm_memory::{FileOffset, GuestMemoryMmap, GuestRegionMmap};
    use vmm_sys_util::tempfile::TempFile;

    use crate::vhost::vhost_user::block::{
        VhostUserBlock, VhostUserConfigFlags, VirtioBlockConfig, VHOST_USER_CONFIG_OFFSET,
    };
    use crate::vhost::vhost_user::test_utils::*;
    use crate::{GuestAddress, VirtioDevice, VirtioDeviceConfig, VirtioQueueConfig, TYPE_BLOCK};
```